### PR TITLE
✨(grades) add a grades api endpoint

### PR DESCRIPTION
**Pull request by jbpenrath**
Create a server to server endpoint to retrieve the student's grade summary for a course. In this way, we are able to retrieve a student progression from an external service.

### CODE REVIEW

File: grades_api/urls.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from django.conf import settings
+from django.conf.urls import patterns, url
+
+urlpatterns = patterns(
+    "grades_api.api",
+    url(r'^{}/(?P<student_name>[\w.+-]+)/$'.format(settings.COURSE_KEY_PATTERN), 'get_student_course_grade', name="student_grade")
```
Comment (sampaccoud) [inline grades_api/urls.py]: or username?
```suggestion
    url(r'^{}/(?P<student_name>[\w.+-]+)/$'.format(settings.COURSE_KEY_PATTERN), 'get_student_course_grade', name="student_grade")
```

File: grades_api/api.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+from django.contrib.auth.models import User
+from django.db import transaction
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from courseware import courses
+from courseware.grades import grade
+
+def _computed_passed(grade_cutoffs, percent = 0.0):
+    """
+    Computes and returns whether the given percent value
+    is a passing grade according to the given grade cutoffs.
```
Comment (sampaccoud) [inline grades_api/api.py:13]: I don't get the point of a default percentage in args?

File: grades_api/api.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    """
+    nonzero_cutoffs = [cutoff for cutoff in grade_cutoffs.values() if cutoff > 0]
+    success_cutoff = min(nonzero_cutoffs) if nonzero_cutoffs else None
+    return success_cutoff and percent >= success_cutoff
```
Comment (sampaccoud) [inline grades_api/api.py]: Not sure about priority between "and" and ">=":
```suggestion
    return success_cutoff and (percent >= success_cutoff)
```

File: grades_api/api.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        student = User.objects.get(username=student_name)
+        course_key = CourseKey.from_string(course_key_string)
+        course = courses.get_course(course_key)
+    except (User.DoesNotExist, ValueError) as error:
+        return Response(str(error), status=404)
+
+    grade_summary = grade(student, request, course, keep_raw_scores=False)
+    grade_summary.update({'passed':_computed_passed(course.grade_cutoffs, grade_summary.get('percent'))})
+
+    return Response(grade_summary, status=200)
```
Comment (sampaccoud) [inline grades_api/api.py]: No newline at end of file

**Review by sampaccoud — APPROVED** [review]

File: grades_api/api.py (hunk 2 above)
Comment (jbpenrath) [reply grades_api/api.py:13]: Mostly to help reader to understand that percent should be a float value.
If you prefer I can remove this default value and provide to the `grade_summary.get('percent')` a default value

Comment (sampaccoud) [reply grades_api/api.py:13]: I would say it is a required value. Does it make sense with the way you use it?

Comment (jbpenrath) [reply grades_api/api.py:13]: After a quick check, grade_summary['percent'] seems to be always a float, so I do not need to define a default value.

Comment (sampaccoud) [reply grades_api/api.py:13]: Ok Let's merge it :tada:

**Pull request merged**